include/aws/crt/StringView.h

```diff
@@ -0,0 +1,136 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#if __cplusplus >= 201703L
+#    include <string_view>
+#else
+#    include <algorithm>
+#    include <cassert>
+#    include <iterator>
+#    include <type_traits>
+#endif
+
+namespace Aws
+{
+    namespace Crt
+    {
+#if __cplusplus >= 201703L
+        using StringView = std::string_view;
+#else
+        class StringView
+        {
+          public:
+            typedef char *pointer;
+            typedef const char *const_pointer;
+            typedef char &reference;
+            typedef const char &const_reference;
+            typedef const_pointer const_iterator;
+            typedef const_iterator iterator;
+            typedef std::reverse_iterator<const_iterator> const_reverse_iterator;
+            typedef const_reverse_iterator reverse_iterator;
+            static constexpr const size_t npos = static_cast<size_t>(-1);
+
+            StringView() noexcept : m_size(0), m_data(nullptr) {}
+            StringView(const StringView &other) noexcept = default;
+            StringView(const char *s, size_t size) noexcept : m_size(size), m_data(s) {}
+            StringView(const char *s) noexcept : m_size(std::char_traits<char>::length(s)), m_data(s) {}
+
+            StringView &operator=(StringView &other) noexcept = default;
+
+            const_iterator begin() const noexcept { return cbegin(); }
+
+            const_iterator end() const noexcept { return cend(); }
+
+            const_iterator cbegin() const noexcept { return m_data; }
+
+            const_iterator cend() const noexcept { return m_data + m_size; }
+
+            const_reverse_iterator rbegin() const noexcept { return const_reverse_iterator(cend()); }
+
+            const_reverse_iterator rend() const noexcept { return const_reverse_iterator(cbegin()); }
+
+            const_reverse_iterator crbegin() const noexcept { return const_reverse_iterator(cend()); }
+
+            const_reverse_iterator crend() const noexcept { return const_reverse_iterator(cbegin()); }
+
+            size_t size() const noexcept { return m_size; }
+
+            size_t length() const noexcept { return m_size; }
+
+            size_t max_size() const noexcept { return (std::numeric_limits<size_t>::max)(); }
+
+            bool empty() const noexcept { return m_size == 0; }
+
+            const_reference operator[](size_t pos) const noexcept
```

Comment:
operator[] is NOT noexcept [link](https://en.cppreference.com/w/cpp/string/basic_string_view/operator_at)

Reply:
Those NOT noexcept functions throw in standard implementation. Here in crt-cpp we don't use throw.